validator/sawtooth_validator/server/core.py

```diff
@@ -240,7 +241,10 @@ def __init__(self,
             max_incoming_connections=20,
             max_future_callback_workers=10)
 
-        consensus_notifier = ConsensusNotifier(consensus_service)
+        registered_engines = ConcurrentMultiMap()
```

Comment:
Can we make this a separate class and give the operations on it descriptive names? For example, `ConsensusRegistry` with methods `register_engine()`, `unregister_engine()`, `get_connection_id_by_engine_info()`, and `get_engine_info_by_connection_id()`.